src/Mod/TechDraw/Gui/QGIViewDimension.cpp

```diff
@@ -351,21 +351,30 @@ void QGIDatumLabel::hoverLeaveEvent(QGraphicsSceneHoverEvent* event)
 
 QRectF QGIDatumLabel::boundingRect() const
 {
-    return childrenBoundingRect();
+    QRectF totalRect;
+    for (QGraphicsItem* item : m_textItems->childItems()) {
+        if (QGCustomText* customText = dynamic_cast<QGCustomText*>(item)) {
+            QRectF itemRect = customText->alignmentRect();
+            QPointF pos = customText->pos();
+            itemRect.translate(pos.x(), pos.y());
+            if (totalRect.isNull()) {
+                totalRect = itemRect;
+            } else {
+                totalRect = totalRect.united(itemRect);
+            }
```

Comment:
How about using the ternary operator?
```suggestion
            totalRect = totalRect.isNull() ? itemRect : totalRect.united(itemRect);
```

Reply:
Incorporated.